### ISSUE TITLE
appstream 0.16.3 build failure

### ISSUE THREAD

**Issue by chenrui333** [ximion/appstream#552]
👋 While trying to package appstream 0.16.3 for Homebrew, I ran into some build failure as below:

```
  ../src/as-system-info.c:469:23: error: use of undeclared identifier 'DT_LNK'
                          if (ent->d_type == DT_LNK)
                                             ^
  ../src/as-system-info.c:472:23: error: use of undeclared identifier 'DT_DIR'
                          if (ent->d_type == DT_DIR) {
                                             ^
  2 errors generated.
```

This is the full homebrew build logs, https://gist.github.com/chenrui333/dd1d0b21489e7d5fa5bcf3ab1857db22
this is the CI build, https://github.com/Homebrew/homebrew-core/actions/runs/6756619124/job/18366307606

relates to https://github.com/Homebrew/homebrew-core/pull/142989

Thanks!

**Comment by danyeaw** [comment]
Hi @chenrui333, I'm also trying to get libadwaita 1.4.0 to build. I think this error is already fixed in 8d752b0637960c8e31a325e09f3c2c730ee5bd86. If you add this to your formula it should resolve it:

```
  patch do
    url "https://github.com/ximion/appstream/commit/8d752b0637960c8e31a325e09f3c2c730ee5bd86.patch?full_index=1"
    sha256 "3302241cb1c935a5b601ee490e3c0d70a1ae85ab7ad285de66598bdacaf6f7b4"
  end
```

**Comment by chenrui333** [comment]
Should be good now with https://github.com/Homebrew/homebrew-core/pull/153408
